packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -439,77 +440,125 @@ describe("SharedTree benchmarks", () => {
-								insert(receiver.kernel.checkout, 0, `r${iCommit}`);
-							}
-							// These are the commits that should be bunched together
-							for (let iCommit = 0; iCommit < bunchSize; iCommit++) {
-								insert(sender.kernel.checkout, 0, `s${iCommit}`);
-							}
-							// Ensure the sender has sent the ops
-							provider.processMessages();
-							// Prevent the sender from receiving anything else since we only want to measure the rebase on the receiver
-							sender.setConnected(false);
-							const before = state.timer.now();
-							// Allow the receiver to receive the bunched commits.
-							// This should force the local branch to be rebased over the bunch.
-							receiver.setConnected(true);
-							const after = state.timer.now();
-							duration = state.timer.toSeconds(before, after);
-						} while (state.recordBatch(duration));
-					},
-					// Force batch size of 1
-					minBatchDurationSeconds: 0,
-				});
-				if (!isInPerformanceTestingMode) {
-					test.timeout(5000);
-				}
+	describe("Op Bunching", () => {
+		/**
+		 * Helper function to setup the sender and receiver trees for the op bunching benchmarks.
+		 * @remarks It pauses the processing of both sender and receiver to enable the test to control when the
+		 * messages are processed.
+		 */
+		function setupSenderAndReceiver() {
+			const provider = new TestTreeProviderLite(
+				2,
+				factory,
+				undefined /* useDeterministicSessionIds */,
+				FlushMode.TurnBased,
+			);
+			const sender = provider.trees[0];
+			const receiver = provider.trees[1];
+			sender.containerRuntime.pauseProcessing();
+			receiver.containerRuntime.pauseProcessing();
+			return { provider, sender, receiver };
+		}
+
+		/**
+		 * Helper function to send local commits from a given tree.
+		 * In turn based flush mode, the messages won't be sequenced until flush is called explicitly
+		 * on the tree's container runtime.
+		 * In Immediate flush mode, the messages will be flushed and sequenced immediately. So, this
+		 * function will behave similar to 'sequenceLocalCommits'.
+		 */
+		function sendLocalCommits(
+			tree: SharedTreeWithContainerRuntime,
+			count: number,
+			commitPrefix: string,
+		) {
+			for (let iCommit = 0; iCommit < count; iCommit++) {
+				insert(tree.kernel.checkout, 0, `${commitPrefix}${iCommit}`);
 			}
 		}
+
+		/**
+		 * Helper function that sends and sequences local commits from a given tree.
+		 */
+		function sequenceLocalCommits(
+			tree: SharedTreeWithContainerRuntime,
+			count: number,
+			commitPrefix: string,
+		) {
+			sendLocalCommits(tree, count, commitPrefix);
+			tree.containerRuntime.flush();
+		}
+
+		/**
+		 * Helper function that processes all sequenced commits on a given tree.
+		 */
+		function receiveSequencedCommits(
+			tree: SharedTreeWithContainerRuntime,
+			provider: TestTreeProviderLite,
+		) {
+			tree.containerRuntime.resumeProcessing();
+			provider.processMessages(false /* flush */);
```

Comment:
This helper's implementation is now simple enough that I'd advocate just inlining the code because the code is not much longer and its effects are much clearer. For example, when calling the helper, it's not clear whether the helper will call resumeProcessing or set connected = true before processing the messages, and it's not clear whether it will restore the paused/unpaused and connected/disconnected status of the given tree before returning. You could describe that contract in the function doc comment, but it seems more straightforward just to replicate the code. I see this as a good thing: you've made the underlying functionality nice enough that the helpers are no longer adding value.

Reply:
The helper calls processMessages on the `MockContainerRuntimeFactory` not the `MockContainerRuntim`. It's a little confusing that it calls on the `MockContainerRuntimeFactory` but it handles the respondibility of the service as well. So, essentially, we are telling the service to process the messages it has reeceived from all the clients.

So, the helper is essentially doing the following:
- Asks all the clients (`MockContainerRuntime`) to submit the messages to service (`MockContainerRuntimeFactory`) for sequencing. This is only useful in `TurnBased` mode.
- Asks the service to then send those messages to the clients for processing.

There is no simple way for the tests to do this directly. We can do it but it's not worth it IMO.

Reply:
I'm so confused. The helpers I'm referring to are `receiveSequencedCommits` and `sequenceLocalCommits`. Is that what your're also talking about?

Reply:
Oh, I was talking about the `processMessages` helper. You are right that `receiveSequencedCommits` and `sequenceLocalCommits` are simple enough to not require a helper.

Reply:
The helper seemed a little more useful when I add more usages in https://github.com/microsoft/FluidFramework/pull/24093. However, you are right that it's not apparent what is happening under the hood. Comments explains it but inlining them is fine too.